Bottom: 6bf16374f9b8f6286e09ee73bcfffb12a0c53eab
Top:    4fa1c50e2dd445d88dac3dfe5d57a5bf9550a108
Author: Pavel Roskin <proski@gnu.org>
Date:   2013-11-04 10:30:55 -0500

Refresh of initial

---

diff --git a/farm-checkout b/farm-checkout
index 0ac1444..ca91b3e 100755
--- a/farm-checkout
+++ b/farm-checkout
@@ -1,14 +1,22 @@
-#! /bin/sh
+#! /bin/bash
 
 set -e
-source ~/.farmrc
+. $HOME/.farmrc
 
 cd $HOME/src/linux
 mkdir -p $farm
 for tag in $tags; do
 	srctree=$farm/`echo $tag | sed 's,/,-,'`
-	test -d $srctree && continue
+	echo -n "Checking out $srctree "
+	if test -d $srctree; then
+		echo -e "\e[33mEXISTS\e[0m"
+		continue
+	fi
 	mkdir $srctree
-	echo "Checking out $srctree"
-	git --work-tree=$srctree checkout $tag -- .
+	if git --work-tree=$srctree checkout $tag -- .; then
+		echo -e "\e[32mOK\e[0m"
+	else
+		rm -rf $srctree
+		echo -e "\e[31mERROR\e[0m"
+	fi
 done
diff --git a/farm-clean b/farm-clean
index c7aad8c..1ab0d27 100755
--- a/farm-clean
+++ b/farm-clean
@@ -1,7 +1,7 @@
-#! /bin/sh
+#! /bin/bash
 
 set -e
-source ~/.farmrc
+. $HOME/.farmrc
 
 for tree in $farm/*; do
 	test -d $tree/Documentation || continue
diff --git a/farm-compile b/farm-compile
index 32196dd..1b8f8aa 100755
--- a/farm-compile
+++ b/farm-compile
@@ -1,7 +1,7 @@
-#! /bin/sh
+#! /bin/bash
 
 set -e
-source ~/.farmrc
+. $HOME/.farmrc
 
 # MadWifi
 if test -d tools; then
diff --git a/farm-config32 b/farm-config32
index 55c390a..c1604fc 100755
--- a/farm-config32
+++ b/farm-config32
@@ -1,7 +1,7 @@
-#! /bin/sh
+#! /bin/bash
 
 set -e
-source ~/.farmrc
+. $HOME/.farmrc
 
 arch=i386
 ccflags=-m32
diff --git a/farm-config64 b/farm-config64
index 7e8b82d..b4dc14b 100755
--- a/farm-config64
+++ b/farm-config64
@@ -1,7 +1,7 @@
-#! /bin/sh
+#! /bin/bash
 
 set -e
-source ~/.farmrc
+. $HOME/.farmrc
 
 arch=x86_64
 ccflags=-m64
diff --git a/farm-hardlink b/farm-hardlink
index 11054d1..b7273e6 100755
--- a/farm-hardlink
+++ b/farm-hardlink
@@ -1,6 +1,6 @@
-#! /bin/sh
+#! /bin/bash
 
 set -e
-source ~/.farmrc
+. $HOME/.farmrc
 
 hardlink -t $farm
diff --git a/farm-warnings b/farm-warnings
index 83d959c..8cbe679 100755
--- a/farm-warnings
+++ b/farm-warnings
@@ -1,3 +1,3 @@
-#! /bin/sh
+#! /bin/bash
 
 grep -E -h -i '(error|warning):' Log* | grep -v /Module.symvers |sort -u
